### PR TITLE
[ampl-mp] Update version, separate port ampl-asl and fix arm build

### PR DESCRIPTION
**Pull request by JackBoosY**
- Since the component asl in the upstream source is separated, the component is separated as a new port `ampl-asl`
- fix arm build
- update version to latest commit

Related to https://github.com/microsoft/vcpkg/pull/14245

### CODE REVIEW

File: ports/ampl-asl/CONTROL [hunk 1]
```diff
@@ -0,0 +1,5 @@
+Source: ampl-asl
+Version: 2020-11-11
+Homepage: https://github.com/ampl/asl
+Description: AMPL Solver Library
+Supports: !uwp
```
Comment (JackBoosY) [inline ports/ampl-asl/CONTROL:5]: Some libraries doesn't support uwp.
```
F:\vcpkg\buildtrees\ampl-asl\src\26eb22e748-4a63c4138c\src\solvers2\func_add.c(440): error C2065: '_pclose': undeclared identifier
```
See https://docs.microsoft.com/en-us/cpp/cppcx/crt-functions-not-supported-in-universal-windows-platform-apps?view=msvc-160

File: ports/ampl-asl/install-extra-headers.patch [hunk 2]
```diff
@@ -0,0 +1,12 @@
+diff --git a/CMakeLists.txt b/CMakeLists.txt
+index 35d248d..8a85aa7 100644
+--- a/CMakeLists.txt
++++ b/CMakeLists.txt
+@@ -317,6 +317,7 @@ if(BUILD_CPP)
+ endif()
+ 
+ install(FILES ${ASL_HEADERS} ${ASL_SOURCE_DIR}/opcode.hd ${ASL_SOURCE_DIR}/r_opn.hd 
++    ${GENERATED_INCLUDE_DIR}/stdio1.h ${GENERATED_INCLUDE_DIR}/arith.h
```
Comment (JackBoosY) [inline ports/ampl-asl/install-extra-headers.patch:9]: Already open a PR to upstream: https://github.com/ampl/asl/pull/4

File: ports/ampl-asl/fix-crt-linkage.patch [hunk 3]
```diff
@@ -0,0 +1,13 @@
+diff --git a/CMakeLists.txt b/CMakeLists.txt
+index 976af78..35d248d 100644
+--- a/CMakeLists.txt
++++ b/CMakeLists.txt
+@@ -256,7 +256,7 @@ macro(createSingleASL name sourcedir sources)
+   endif()
+   if(MSVC)
+     target_compile_options(${name} PRIVATE 
+-      /wd4013 /wd4018 /wd4101 /wd4244 /wd4273 /wd4267 /wd4996 /MT$<$<CONFIG:Debug>:d>)
++      /wd4013 /wd4018 /wd4101 /wd4244 /wd4273 /wd4267 /wd4996)
```
Comment (JackBoosY) [inline ports/ampl-asl/fix-crt-linkage.patch:10]: Use CRT linkage in `C_FLAGS`/`CXX_FLAGS` set by vcpkg.

File: ports/ampl-asl/portfile.cmake [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+
+vcpkg_install_cmake()
+
+vcpkg_copy_pdbs()
+
+vcpkg_fixup_cmake_targets(CONFIG_PATH share/unofficial-asl TARGET_PATH share/unofficial-asl)
+
+file(REMOVE_RECURSE ${CURRENT_PACKAGES_DIR}/debug/include)
+
+file(INSTALL ${CURRENT_PORT_DIR}/copyright DESTINATION ${CURRENT_PACKAGES_DIR}/share/${PORT})
```
Comment (JackBoosY) [inline ports/ampl-asl/portfile.cmake:36]:
```suggestion
# from ampl-mp license
file(INSTALL ${CURRENT_PORT_DIR}/copyright DESTINATION ${CURRENT_PACKAGES_DIR}/share/${PORT})
```

**Review by PhoebeHui — APPROVED** [review]

**Pull request merged**